GenumKit/Stencil/Filters.swift

```diff
@@ -114,12 +114,26 @@ extension StringFilters {
 }
 
 struct ArrayFilters {
-  static func join(_ value: Any?) throws -> Any? {
-    guard let array = value as? [Any] else { throw FilterError.invalidInputType }
-    let strings = array.flatMap { $0 as? String }
-    guard array.count == strings.count else { throw FilterError.invalidInputType }
+  /// Joins an array of values with a separator. By default, it uses `, `
+  /// but you can pass in a different value as an argument.
+  static func join(value: Any?, arguments: [Any?]) throws -> Any? {
```

Comment:
@kylef Wouldn't it be useful to integrate such a filter in Stencil itself in the future?